server.go

```diff
@@ -341,3 +343,32 @@ func Close() {
 	servers = nil
 	m.Unlock()
 }
+
+// gracefulHandler is used by GracefulServer to prevent calling ServeHTTP on
+// to be closed kept-alive connections during the server shutdown.
+type gracefulHandler struct {
```

Comment:
I think this can be deleted now. Is that right? The listener should drop the connection and implement the behavior we want without overriding the handler.